src/StmtToViz.cpp

```diff
@@ -0,0 +1,2389 @@
+    }
+    string symbol(const string &x) {
+        return span("Symbol", x);
+    }
+
+    Scope<int> scope;
+    string var(const string &x) {
+        int id;
+        if (scope.contains(x)) {
+            id = scope.get(x);
+        } else {
+            id = unique_id();
+            scope.push(x, id);
+        }
+
+        stringstream s;
+        s << "<b class='Variable Matched' id='" << id << "-" << unique_id() << "'>";
+        s << x;
+        s << "</b>";
+        return s.str();
+    }
+
+    void print_list(const std::vector<Expr> &args) {
+        for (size_t i = 0; i < args.size(); i++) {
+            if (i > 0) {
+                stream << matched(",") << " ";
+            }
+            print(args[i]);
+        }
+    }
+    void print_list(const string &l, const std::vector<Expr> &args, const string &r) {
+        stream << matched(l);
+        print_list(args);
+        stream << matched(r);
+    }
+
+    string open_expand_button(int id) {
+        stringstream button;
+        button << "<a class=ExpandButton onclick='return toggle(" << id << ", " << tooltip_count
+               << ");'>"
+               << "<div style='position:relative; width:0; height:0;'>"
+               << "<div class=ShowHide style='display:none;' id=" << id << "-show"
+               << "><i class='fa fa-plus-square-o'></i></div>"
+               << "<div class=ShowHide id=" << id << "-hide"
+               << "><i class='fa fa-minus-square-o'></i></div>"
+               << "</div>";
+        return button.str();
+    }
+
+    string close_expand_button() {
+        return "</a>";
+    }
+
+    void visit(const IntImm *op) override {
+        stream << open_span("IntImm Imm");
+        stream << Expr(op);
+        stream << close_span();
+    }
+
+    void visit(const UIntImm *op) override {
+        stream << open_span("UIntImm Imm");
+        stream << Expr(op);
+        stream << close_span();
+    }
+
+    void visit(const FloatImm *op) override {
+        stream << open_span("FloatImm Imm");
+        stream << Expr(op);
+        stream << close_span();
+    }
+
+    void visit(const StringImm *op) override {
+        stream << open_span("StringImm");
+        stream << "\"";
+        for (unsigned char c : op->value) {
+            if (c >= ' ' && c <= '~' && c != '\\' && c != '"') {
+                stream << c;
+            } else {
+                stream << "\\";
+                switch (c) {
+                case '"':
+                    stream << "\"";
+                    break;
+                case '\\':
+                    stream << "\\";
+                    break;
+                case '\t':
+                    stream << "t";
+                    break;
+                case '\r':
+                    stream << "r";
+                    break;
+                case '\n':
+                    stream << "n";
+                    break;
+                default:
+                    string hex_digits = "0123456789ABCDEF";
```

Comment:
This is creating (and destroying) this string for *every character* you output. It should really be a `static const char [16]`.